numpy/lib/recfunctions.py

```diff
@@ -1081,7 +1081,9 @@ def apply_along_fields(func, arr):
     Apply function 'func' as a reduction across fields of a structured array.
 
     This is similar to `apply_along_axis`, but treats the fields of a
-    structured array as an extra axis.
+    structured array as an extra axis. The fields are all first cast to a
+    common type following the type-promotion rules from `np.result_type`
```

Comment:
I don't think this will link correctly - you need to use `numpy.result_type` or just `result_type`